core/src/plan/expr/function.rs

```diff
@@ -1,17 +1,18 @@
 use {
     crate::ast::{Expr, Function},
-    std::iter::empty,
+    std::iter::{empty, once},
 };
 
 impl Function {
-    pub fn as_exprs(&self) -> impl ExactSizeIterator<Item = &Expr> {
-        #[derive(iter_enum::Iterator, iter_enum::ExactSizeIterator)]
-        enum Exprs<I0, I1, I2, I3, I4> {
+    pub fn as_exprs(&self) -> impl Iterator<Item = &Expr> {
+        #[derive(iter_enum::Iterator)]
+        enum Exprs<I0, I1, I2, I3, I4, I5> {
             Empty(I0),
             Single(I1),
             Double(I2),
             Triple(I3),
             VariableArgs(I4),
+            TypedStringVariableArgs(I5),
```

Comment:
This looks interesting that we need another name for variable args with a single expression.

nit suggestion for the name.
Though these are internal variables so it does not need to be named too carefully..
We have a variant for `ast::Expr::TypedString` and `TypedString` keyword here is not related to `Expr::TypedString` so it could be little bit confusing.
Then.. how about simply `VariableArgs2` or `VariableArgsWithSingle`...?